pkgs/tools/compression/ouch/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ouch";
+  version = "0.1.5";
+
+  src = fetchFromGitHub {
+    owner = "vrmiguel";
+    repo = pname;
+    rev = version;
+    sha256 = "00ah8hgrppa61jhwb74zl5b509q0yp2pp27w9frm814iqx70qn38";
+  };
+
+  cargoPatches = [
+    # a patch file to add Cargo.lock in the source code
+    # https://github.com/vrmiguel/ouch/pull/46
+    ./add-Cargo.lock.patch
+  ];
```

Comment:
Please use cargoLock instead of a patch https://github.com/NixOS/nixpkgs/blob/d9f9e5d2088c76aea3244a40c40612ff789576f2/pkgs/build-support/rust/test/import-cargo-lock/basic/default.nix#L9.

Reply:
@SuperSandro2000 I did try that, but it gives me this error:

```
$ nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'                                                                                                
error: cannot coerce a set to a string, at /nix/store/mzh83bazd07785nibm8swwhg2cvi58zi-nixos-21.05.1555.6b940846e99/nixos/pkgs/stdenv/generic/make-derivation.nix:201:11
(use '--show-trace' to show detailed location information)
```

Am I doing something wrong ?

Reply:
Did you quote ./Cargo.lock ? It shouldn't be quoted. Can you post your diff? Than I can help you better with it.

Reply:
@SuperSandro2000 I didn't quote it, but I have pushed a commit now to see if it passes on the CI (I hope this doesn't involve changes in the hashes!). I have a feeling if this is because the <nixpkgs> in my system is old (I'm using 21.05 based one and it seems the cargo lock feature was introduced couple of months ago).

Reply:
Seems the build is failing but because of a different issue: https://github.com/NixOS/nixpkgs/pull/131538/checks?check_run_id=3193556591

Reply:
@SuperSandro2000 Based on our discussion in Matrix, I have updated this to use `cargoPatches` now as `cargoLock` doesn't seem to work currently. The CI seems to pass too. Thanks for all the help!